updated search docs

winzamark123 opened the pull request.

VineeTagarwaL-code commented:
1. The content is only string, image_url, links etc are not inside of content
<img width="582" height="117" alt="Screenshot 2025-07-31 at 3 17 51 AM" src="https://github.com/user-attachments/assets/af60eaab-a186-44ac-9ca3-619612b07916" />
2. add snippets are a part of content object
3. for objects we should have the object response
<img width="613" height="101" alt="Screenshot 2025-07-31 at 3 23 57 AM" src="https://github.com/user-attachments/assets/456d1547-5ba2-4cb8-89b3-4cea9ca2903a" />
4. for optional keys ( we should have a tag that says optional )

winzamark123 commented:
> 1. The content is only string, image_url, links etc are not inside of content
> 
> <img alt="Screenshot 2025-07-31 at 3 17 51 AM" width="582" height="117" src="https://private-user-images.githubusercontent.com/91052168/472725461-af60eaab-a186-44ac-9ca3-619612b07916.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM5MTI4MjgsIm5iZiI6MTc1MzkxMjUyOCwicGF0aCI6Ii85MTA1MjE2OC80NzI3MjU0NjEtYWY2MGVhYWItYTE4Ni00NGFjLTljYTMtNjE5NjEyYjA3OTE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzMwVDIxNTUyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZiYmM5OGQyNjdmM2Y2ODI2MTA3ODgyZmIzYTI0ZmQ1ZGEyOTMyOWM5MDNkYzc3NzI2YTU2MDRlZjI0MGM0ODQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QnpKFzWJlyXnMJdMkn9hpprisvmkkZR7_Cle2ljgYZA">
> 2. add snippets are a part of content object
> 3. for objects we should have the object response
> 
> <img alt="Screenshot 2025-07-31 at 3 23 57 AM" width="613" height="101" src="https://private-user-images.githubusercontent.com/91052168/472727151-456d1547-5ba2-4cb8-89b3-4cea9ca2903a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM5MTI4MjgsIm5iZiI6MTc1MzkxMjUyOCwicGF0aCI6Ii85MTA1MjE2OC80NzI3MjcxNTEtNDU2ZDE1NDctNWJhMi00Y2I4LTg5YjMtNGNlYTljYTI5MDNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzMwVDIxNTUyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyMzk3NTJiY2QyMTFkYmVhYmYzZDJiMjYwOGM5ZGI0NjIyNWFjYTA2M2MyODdlM2Q3MTc2ZDM5ZTA1Y2Q3ZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZM6XBZ4fkTKO-SRwe_AbnLZxnlwH-T2RWVub7LGdQNU">
> 4. for optional keys ( we should have a tag that says optional )

1. removed image_uurls and links content seems to only be string
2. snippets are already part of the response but not part of the content
3. added object response for geoloc
4. currently we only do required fields and no "optional" everything defaults to optional unless specified as "required"
